### PR TITLE
feat: create span question responses

### LINKED ISSUE #45: [TASK] Allow to create responses for span questions

### PR DESCRIPTION
**Pull request by jfcalvo**
# Description

This PR add changes so we can create responses for span questions.

Closes #45

**Type of change**

(Please delete options that are not relevant. Remember to title the PR according to the type of change)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (change restructuring the codebase without changing functionality)
- [ ] Improvement (change adding some improvement to an existing functionality)
- [ ] Documentation update

**How Has This Been Tested**

(Please describe the tests that you ran to verify your changes. And ideally, reference `tests`)

- [x] Adding new tests.

**Checklist**

- [ ] I added relevant documentation
- [ ] follows the style guidelines of this project
- [ ] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I filled out [the contributor form](https://tally.so/r/n9XrxK) (see text above)
- [ ] I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

### CODE REVIEW

File: src/argilla_server/models/questions.py [hunk 1]
```diff
@@ -12,17 +12,21 @@
...
-from argilla_server.pydantic_v1 import BaseModel, Field
+from argilla_server.pydantic_v1 import BaseModel, Field, root_validator
 
 try:
     from typing import Annotated
 except ImportError:
     from typing_extensions import Annotated
 
 
+SPAN_QUESTION_RESPONSE_VALUE_MIN_ITEMS = 1
```
Comment (frascuchon) [inline src/argilla_server/models/questions.py:26]: We should accept an empty spans list, which means that no entities were found for the provided text 
```suggestion
SPAN_QUESTION_RESPONSE_VALUE_MIN_ITEMS = 0
```

Comment (jfcalvo) [reply src/argilla_server/models/questions.py]: Done.

File: src/argilla_server/models/questions.py [hunk 2]
```diff
@@ -151,13 +154,57 @@ def check_response(self, response: ResponseValue, status: Optional[ResponseStatu
...
+        return values
+
+
+class SpanQuestionResponseValue(BaseModel):
+    value: List[SpanQuestionResponseValueItem] = Field(..., max_items=SPAN_QUESTION_RESPONSE_VALUE_MAX_ITEMS)
+
+
+class SpanQuestionSettings(BaseQuestionSettings):
+    type: Literal[QuestionType.span]
+    options: List[ValueTextQuestionSettingsOption]
```
Comment (frascuchon) [inline src/argilla_server/models/questions.py:178]:
```suggestion
    options: List[ValueTextQuestionSettingsOption]
    field: str # <- ??
```

Comment (jfcalvo) [reply src/argilla_server/models/questions.py:178]: I'm not using it to check responses that's why I didn't add it. Additionally the handler schemas are already checking that it's there and we are validating that the field is valid in the question creation.

**Review by frascuchon — APPROVED** [review]

**Pull request merged**